src/bmpcodec.c

```diff
@@ -1197,13 +1197,15 @@ gdip_read_bmp_image (void *pointer, GpImage **image, ImageSource source)
 			status = gdip_read_bmp_indexed (pointer, pixels, upsidedown, originalFormat, originalStride, result->active_bitmap->width, result->active_bitmap->height, source);
 			if (status != Ok) {
 				gdip_bitmap_dispose (result);
+				GdipFree(pixels);
 				return status;
 			}
 		}
 	} else {
 		status = gdip_read_bmp_scans (pointer, pixels, upsidedown, originalFormat, originalStride, result->active_bitmap->stride, result->active_bitmap->width, result->active_bitmap->height, source);
 		if (status != Ok) {
 			gdip_bitmap_dispose (result);
+			GdipFree(pixels);
```

Comment:
FWIW I had to revert these two changes as they caused the testbitmap test to crash in master.